pkg/kube/ingress.go

```diff
@@ -0,0 +1,238 @@
+
+	if _, err := ingressInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    ingressWatcher.IngressHandlerAddFunc,
+		UpdateFunc: ingressWatcher.IngressHandlerUpdateFunc,
+		DeleteFunc: ingressWatcher.IngressHandlerDeleteFunc,
+	}); err != nil {
+		return nil, err
+	}
+
+	return ingressWatcher, nil
+}
+
+func (iw *IngressWatcher) Start() error {
+	iw.logger.Debug("Starting IngressWatcher")
+	iw.factory.Start(iw.ctx.Done())
+
+	if !cache.WaitForCacheSync(iw.ctx.Done(), iw.informer.HasSynced) {
+		return errors.New("failed to sync ingress cache")
+	}
+
+	return nil
+}
+
+func (iw *IngressWatcher) Stop() {
+	iw.logger.Debug("Stopping IngressWatcher")
+	iw.factory.Shutdown()
+}
+
+// --- ResourceEventHandler methods ---
+
+func (iw *IngressWatcher) IngressHandlerAddFunc(obj interface{}) {
+	host, path, targets, err := iw.extractIngressValuesFromIngressResource(obj)
+	if err != nil {
+		iw.logger.WarnWith("Add ingress handler failure", "error", err)
+		return
+	}
+
+	if err = iw.ingressCache.Set(host, path, targets); err != nil {
+		iw.logger.WarnWith("Add ingress handler failure- failed to add the new value", "error", err, "object", obj)
+		return
+	}
+}
+
+func (iw *IngressWatcher) IngressHandlerUpdateFunc(oldObj, newObj interface{}) {
+	oldHost, oldPath, oldTargets, err := iw.extractIngressValuesFromIngressResource(oldObj)
+	if err != nil {
+		iw.logger.WarnWith("Update ingress handler - failed to extract values from old object", "error", err)
+		return
+	}
+
+	newHost, newPath, newTargets, err := iw.extractIngressValuesFromIngressResource(newObj)
+	if err != nil {
+		iw.logger.WarnWith("Update ingress handler - failed to extract values from new object", "error", err)
+		return
+	}
+
+	// if the host or path has changed, we need to delete the old entry
+	if oldHost != newHost || oldPath != newPath {
+		if err = iw.ingressCache.Delete(oldHost, oldPath, oldTargets); err != nil {
+			iw.logger.WarnWith("Update ingress handler failure - failed to delete old ingress", "error", err)
+		}
+	}
+
+	if err = iw.ingressCache.Set(newHost, newPath, newTargets); err != nil {
+		iw.logger.WarnWith("Update ingress handler failure- failed to add the new value", "error", err, "object", newObj)
+		return
+	}
+}
+
+func (iw *IngressWatcher) IngressHandlerDeleteFunc(obj interface{}) {
+	host, path, targets, err := iw.extractIngressValuesFromIngressResource(obj)
+	if err != nil {
+		iw.logger.WarnWith("Delete ingress handler failure- failed to extract values from object", "error", err)
+		return
+	}
+
+	if err = iw.ingressCache.Delete(host, path, targets); err != nil {
+		iw.logger.WarnWith("Delete ingress handler failure- failed delete from cache", "error", err, "object", obj)
+		return
+	}
+}
+
+// --- internal methods ---
+
+// extractIngressValuesFromIngressResource extracts the host, path, and targets from the ingress resource.
+func (iw *IngressWatcher) extractIngressValuesFromIngressResource(obj interface{}) (string, string, []string, error) {
```

Comment:
targets is not an ingress value really
```suggestion
func (iw *IngressWatcher) extractValuesFromIngressResource(obj interface{}) (string, string, []string, error) {
```

Reply:
Fixed here - [CR comments](https://github.com/v3io/scaler/pull/75/commits/fc972c82145e64d4022498dac09dbc3cabf6820d)